config/src/main/java/org/opennms/integration/api/xml/ClasspathXmlLoader.java

```diff
@@ -62,13 +63,10 @@ public ClasspathXmlLoader(Class<?> clazzForClassloader, Class<T> objectClazz, St
             try {
                 try (InputStream is = classLoader.getResourceAsStream(subFolder + File.separator +fileName)) {
                     if (is != null) {
-                        final ByteSource byteSource = new ByteSource() {
-                            @Override
-                            public InputStream openStream() {
-                                return is;
-                            }
-                        };
-                        final String xml = byteSource.asCharSource(StandardCharsets.UTF_8).read();
+                        // Read the input stream line by line collecting it into a String
```

Comment:
If we're going to rework this should we skip the string building and pass the `InputStream` directly to the `Unmarshaller`? (Would required update JaxbUtils)

Reply:
Ideally yes, I just tried to make the smallest change possible for this PR and went with the existing approach. I can change JaxbUtils instead if we're ok with that. Can we get rid of the method that takes a `String` altogether (current use is the only  case).

Reply:
I'd add another method so we respect semver.

Reply:
Sounds good 👍 Will add a new method for dealing with the `InputStream`.